game-core/src/main/java/games/strategy/engine/random/PbemDiceRoller.java

```diff
@@ -177,7 +178,6 @@ void notifyError() {
 
     // should only be called if we are not visible
     // should be called from the event thread
-    // wont return until the roll is done.
```

Comment:
This comment is no longer accurate.

Reply:
Derp.  I forgot that this is a _modal_ dialog, so that the above comment is actually correct.  Restored in 3ad8fcffb.

> So instead of waiting for the window to become visible, wouldn't it make more sense to restructure the code so visibility is triggered in the right order?

Because the dialog is modal, `setVisible(true)` does not return until the dialog is closed.  Therefore, `rollInternal()` must be called before `setVisible(true)`.

Without the `waitForWindowToBecomeVisible()` call, the worst that can happen is the roll will be made without any visual feedback (e.g. if MARTI responds before the EDT gets a chance to display the dialog, the dialog will never be displayed because it will have been disposed).  However, because this code exists, I'm going to assume that exact problem happened sometime in the past to trigger its addition.